cleanlab/datalab/datalab.py

```diff
@@ -0,0 +1,702 @@
+            Name of the issue type to summarize. If `None`, summarizes each of the different issue types previously considered in the audit.
+
+        Returns
+        -------
+        summary :
+            DataFrame where each row corresponds to a type of issue, and columns quantify:
+            the number of examples in the dataset estimated to exhibit this type of issue,
+            and the overall severity of the issue across the dataset (via a numeric quality score where lower values indicate that the issue is overall more severe).
+        """
+        if self.issue_summary.empty:
+            raise ValueError(
+                "No issues found in the dataset. "
+                "Call `find_issues` before calling `get_summary`."
+            )
+
+        if issue_name is None:
+            return self.issue_summary
+
+        row_mask = self.issue_summary["issue_type"] == issue_name
+        if not any(row_mask):
+            raise ValueError(f"Issue type {issue_name} not found in the summary.")
+        return self.issue_summary[row_mask].reset_index(drop=True)
+
+    def _add_issue_summary_to_report(self, summary: pd.DataFrame) -> str:
+        return (
+            "Here is a summary of the different kinds of issues found in the data:\n\n"
+            + summary.to_string(index=False)
+            + "\n\n"
+            + "(Note: A lower score indicates a more severe issue across all examples in the dataset.)\n\n\n"
+        )
+
+    def find_issues(
+        self,
+        *,
+        pred_probs: Optional[np.ndarray] = None,
+        features: Optional[npt.NDArray] = None,
+        knn_graph: Optional[csr_matrix] = None,
+        issue_types: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Checks the dataset for all sorts of common issues in real-world data (in both labels and feature values).
+
+        You can use Datalab to find issues in your data, utilizing *any* model you have already trained.
+        This method only interacts with your model via its predictions or embeddings (and other functions thereof).
+        The more of these inputs you provide, the more types of issues Datalab can detect in your dataset/labels.
+        If you provide a subset of these inputs, Datalab will output what insights it can based on the limited information from your model.
+
+        Note
+        ----
+        The issues are saved in the ``self.issues`` attribute, but are not returned.
+
+        Parameters
+        ----------
+        pred_probs :
+            Out-of-sample predicted class probabilities made by the model for every example in the dataset.
+            To best detect label issues, provide this input obtained from the most accurate model you can produce.
+
+        features :
+            Feature embeddings (vector representations) of every example in the dataset.
+
+        knn_graph :
+            Sparse matrix representing similarities between examples in the dataset in a K nearest neighbor graph.
```

Comment:
can we add one more precise descriptive sentence after this one. Don't think users could construct this properly just based on this one-sentence up here.